ouroboros-consensus/test-consensus/Test/Consensus/Mempool.hs

```diff
@@ -831,9 +832,80 @@ prop_TxSeq_lookupByTicketNo_sound smalls small =
     txseq =
         foldl' (TxSeq.:>) TxSeq.Empty $ map mkTicket haystack
 
-    mkTicket x = TxTicket x (mkTicketNo x)
+    mkTicket x = TxTicket x (mkTicketNo x) 0
     mkTicketNo = TicketNo . toEnum
 
+-- | Test that the 'fst' of the result of 'splitAfterTxSize' only contains
+-- 'TxTicket's whose summed up transaction sizes are less than or equal to
+-- that of the 'TxSizeInBytes' which the 'TxSeq' was split on.
+prop_TxSeq_splitAfterTxSize :: TxSizeSplitTestSetup -> Property
+prop_TxSeq_splitAfterTxSize tss =
+      property $ txSizeSum (txTickets before) <= tssTxSizeToSplitOn
+  where
+    TxSizeSplitTestSetup { tssTxSizeToSplitOn } = tss
+
+    (before, _after) = splitAfterTxSize txseq tssTxSizeToSplitOn
+
+    txseq :: TxSeq Int
+    txseq = txSizeSplitTestSetupToTxSeq tss
+
+    txSizeSum :: [TxTicket tx] -> TxSizeInBytes
+    txSizeSum = sum . map txTicketTxSizeInBytes
+
+
+-- | Test that the results of 'splitAfterTxSizeSpec', a specification of
+-- 'splitAfterTxSize', match those of the real 'splitAfterTxSize'
+-- implementation.
+prop_TxSeq_splitAfterTxSizeSpec :: TxSizeSplitTestSetup -> Property
+prop_TxSeq_splitAfterTxSizeSpec tss =
+    splitAndCheck tssTxSizeToSplitOn
+  where
+    TxSizeSplitTestSetup { tssTxSizeToSplitOn } = tss
+
+    txseq :: TxSeq Int
+    txseq = txSizeSplitTestSetupToTxSeq tss
+
+    splitAndCheck :: TxSizeInBytes -> Property
+    splitAndCheck txsz =
+      let (implBefore, implAfter) = splitAfterTxSize txseq txsz
+          (specBefore, specAfter) = splitAfterTxSizeSpec txseq txsz
+      in      txTickets implBefore === txTickets specBefore
+         .&&. txTickets implAfter  === txTickets specAfter
+
+{-------------------------------------------------------------------------------
+  TxSizeSplitTestSetup
+-------------------------------------------------------------------------------}
+
+data TxSizeSplitTestSetup = TxSizeSplitTestSetup
+  { tssTxSizes         :: ![TxSizeInBytes]
+  , tssTxSizeToSplitOn :: !TxSizeInBytes
+  } deriving Show
+
+instance Arbitrary TxSizeSplitTestSetup where
+  arbitrary = do
+    let txSizeMaxBound = 10000000 -- 10MB transaction max bound
+    txSizes <- listOf $ choose (1, txSizeMaxBound)
+    let totalTxSize = sum txSizes
+    txSizeToSplitOn <- choose (0, totalTxSize)
+    pure TxSizeSplitTestSetup
+      { tssTxSizes = txSizes
+      , tssTxSizeToSplitOn = txSizeToSplitOn
+      }
+
+  shrink TxSizeSplitTestSetup { tssTxSizes, tssTxSizeToSplitOn } =
+    [ TxSizeSplitTestSetup
+        { tssTxSizes         = tssTxSizes'
+        , tssTxSizeToSplitOn = tssTxSizeToSplitOn'
+        }
+    | tssTxSizes' <- shrinkList (const []) tssTxSizes
+    , tssTxSizeToSplitOn' <- shrinkIntegral tssTxSizeToSplitOn
```

Comment:
This isn't the most efficient shrinker: what if we found a failure for `sum tssTxSize == tssTxSizeToSplitOn`? Now shrinking will probably generate tons of smaller non-relevant test cases. However, I don't think it's worth improving this (until we get actual test failures that are too large). So you can leave this as is.